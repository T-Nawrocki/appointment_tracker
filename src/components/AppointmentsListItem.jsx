import React from 'react';

const AppointmentsListItem = (props) => {
  return ( 
    <li classname="appointments-list-item" key={props.index}>
      <h2>{props.appointment.title}</h2>
      <p>{props.appointment.time} {props.appointment.date}</p>
    </li>
  );
}
 
export default AppointmentsListItem;
