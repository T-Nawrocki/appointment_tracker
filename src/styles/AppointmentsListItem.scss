@use 'common/colours';

.appointments-list-item {
  display: flex;
  flex-direction: row;
  margin: 1%;
  border: 3px colours.$dark solid;
  border-radius: 25px;
  padding: 1%;
  background: colours.$mid;
  text-align: left;
  
  h4 {
    margin: 0;
  }

  .content {
    flex: 3;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }

  button {
    flex: 1;
    cursor: pointer;
  }

}
