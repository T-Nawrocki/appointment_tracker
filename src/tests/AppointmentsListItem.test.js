import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import AppointmentsListItem from '../components/AppointmentsListItem';

configure({ adapter: new Adapter() });

describe('Appointments List Item', () => {
  it('should render the list item', () => {
    const tree = renderer.create(<AppointmentsListItem appointment={{
      id: 1,
      title: "title",
      dateTime: new Date(2020, 0, 1)
    }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
