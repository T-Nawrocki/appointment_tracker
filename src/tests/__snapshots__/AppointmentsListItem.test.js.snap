// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointments List Item should render the list item 1`] = `
<li
  className="appointments-list-item"
>
  <div
    className="content"
  >
    <h4>
      title
    </h4>
    <div>
      Wednesday, 1 January 2020 at 00:00
    </div>
  </div>
  <button
    className="delete-button"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
